### PR TITLE
Span Creation Optimizations

### PR DESCRIPTION
**Pull request by bengl**
### What does this PR do?
- combine 2 loops into one in span creation
- add a fast path for childOf without references in startSpan

See commit message of first commit for performance difference

### Motivation
Better Performance

### CODE REVIEW

File: packages/dd-trace/src/opentracing/tracer.js [hunk 1]
```diff
@@ -52,12 +54,32 @@ class DatadogTracer extends Tracer {
     }
   }
 
+  startSpan (name, fields) {
+    if (fields && fields.references) {
+      return super.startSpan(name, fields)
+    } else if (fields && fields.childOf) {
+      let parent = fields.childOf
+      if (parent instanceof OTSpan) {
```
Comment (rochdev) [inline packages/dd-trace/src/opentracing/tracer.js:62]: Why check for an OT span specifically? If an OT span is received that is not ours it won't work.

File: packages/dd-trace/src/opentracing/tracer.js [hunk 2]
```diff
@@ -52,12 +54,32 @@ class DatadogTracer extends Tracer {
     }
   }
 
+  startSpan (name, fields) {
+    if (fields && fields.references) {
```
Comment (rochdev) [inline packages/dd-trace/src/opentracing/tracer.js:58]: Condition could be simplified to not repeat the fields check and relying more on returning early before any other conditions.

File: packages/dd-trace/src/opentracing/tracer.js [hunk 3]
```diff
@@ -52,12 +54,32 @@ class DatadogTracer extends Tracer {
...
+      return super.startSpan(name, fields)
+    } else if (fields && fields.childOf) {
+      let parent = fields.childOf
+      if (parent instanceof OTSpan) {
+        parent = parent.context()
+      }
+      if (parent instanceof OTSpanContext) {
```
Comment (rochdev) [inline packages/dd-trace/src/opentracing/tracer.js:65]: Same as above.

**Review by rochdev — CHANGES_REQUESTED** [review]

File: packages/dd-trace/src/opentracing/tracer.js (hunk 3 above)
Comment (bengl) [reply packages/dd-trace/src/opentracing/tracer.js:65]: Same answer as above.

File: packages/dd-trace/src/opentracing/tracer.js (hunk 2 above)
Comment (bengl) [reply packages/dd-trace/src/opentracing/tracer.js]: I simplified this a bit.

File: packages/dd-trace/src/opentracing/tracer.js (hunk 1 above)
Comment (bengl) [reply packages/dd-trace/src/opentracing/tracer.js]: I was under the impression it could be an OT Span. This is now fixed.

**Review by rochdev — APPROVED** [review]

**Pull request merged**